staging/src/k8s.io/apiserver/pkg/registry/generic/registry/storage_factory.go

```diff
@@ -64,7 +64,35 @@ func StorageWithCacher(capacity int) generic.StorageDecorator {
 			cacher.Stop()
 			d()
 		}
-
+		RegisterStorageCleanup(destroyFunc)
```

Comment:
comment: TODO: remove this when we shutdown storages correctly, <link to @frobware's

Reply:
Done

Reply:
comment still missing here